docs/PR_Guidelines.md

```diff
@@ -0,0 +1,13 @@
+# Guidelines for creating a good pull request
+
+1. A PR should describe the change clearly and most importantly it should mention the motivation behind the change. Filling out the PR template should satisfy this guideline.
+2. If the PR is fixing a performance issue, mention the improvement and how the measurement was done (for educational purposes).
+3. Do not leave comments unresolved. If PR comments have been addressed without making the requested code changes, explicitly mark them resolved with an appropriate comment explaining why you're resolving it. If you intend to resolve it in a follow up PR, create a task and mention why this comment cannot be fixed in this PR. Leaving comments unresolved sets a wrong precedent for other contributors that it's ok to ignore comments. 
```

Comment:
Should we have a github board for these tasks?

Reply:
Yeah, good Idea. I think we should. I'll check with Faith on this.

Reply:
Adding @faxu for awareness.